docs/notebooks/expected_improvement.pct.py

```diff
@@ -216,8 +217,10 @@ def build_model(data):
 # Here we use the `BatchMonteCarloExpectedImprovement` function. Note that this acquisition function is computed using a Monte-Carlo method (so it requires a `sample_size`), but with a reparametrisation trick, which makes it deterministic.
 
 # %%
+from trieste.acquisition.rule import BatchAcquisitionRule
+
 qei = trieste.acquisition.BatchMonteCarloExpectedImprovement(sample_size=1000)
-batch_rule = trieste.acquisition.rule.BatchAcquisitionRule(
+batch_rule: BatchAcquisitionRule[Box] = BatchAcquisitionRule(
     num_query_points=3, builder=qei.using(OBJECTIVE)
 )
 
```

Comment:
This looks good for now. But will hopefully be replaced soonish with a singe EGO rule.

Reply:
yeah I'm also not happy with having to type hint these. The fix (short of Python implementing look-ahead type inference ... a name I just made up) is to pass the search space to the acquisition rule constructor, rather than `acquire`. This would mean the search space isn't passed to the `BayesianOptimizer`, Victor says that wouldn't be intuitive for users. Maybe he's right. There are ways round that anyway. I might do that while we're doing this work.

Reply:
moreover, passing the search space to the constructor would probably mean we don't need `SP_contra`

Reply:
True!